frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts

```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
+    const beforeCapacityAdditionDevicesCount = deviceSets[index].count;
+    addCapacity(beforeCapacityAddition.uid, scName);
+    fetchStorageClusterJson().then((res) => {
+      withJSONResult(res, scName, iAndD);
     });
 
-    it('New device set is not created', () =>
-      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets));
-    it('Devices count is incremented by 1 in the corresponding device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].count).toBe(beforeCapacityAddition.devicesCount + 1));
-    it('Osd portability is not modified in the corresponding device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].portable).toBe(beforeCapacityAddition.portability));
+    cy.log('New device set is not created').then(() =>
+      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length),
+    );
+
+    cy.log('Devices count is incremented by 1 in the corresponding device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].count).toBe(beforeCapacityAdditionDevicesCount + 1),
+    );
+
+    cy.log('Osd portability is not modified in the corresponding device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].portable).toBe(beforeCapacityAdditionPortability),
+    );
   });
 
-  describe(`Add capacity with a new storage class having NO-PROVISIONER as provisioner`, () => {
+  it(`Add capacity with a new storage class having NO-PROVISIONER as provisioner`, () => {
     const { name: scName } = testNoProvisionerSC.metadata;
     const iAndD: IndexAndDeviceSet = { index: 0, deviceSets: [] };
-    before(() => {
-      fetchStorageClusterJson().then((res) => {
-        const json: K8sResourceKind = JSON.parse(res.stdout);
-        beforeCapacityAddition.deviceSets = json.spec.storageDeviceSets.length;
-        addCapacity(json.metadata.uid, scName);
-        fetchStorageClusterJson().then((newRes) => {
-          withJSONResult(newRes, scName, iAndD);
-        });
-      });
+    addCapacity(beforeCapacityAddition.uid, scName);
+    fetchStorageClusterJson().then((res) => {
+      withJSONResult(res, scName, iAndD);
     });
 
-    it('New device set is created', () =>
-      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets + 1));
-    it('Device count is 1 in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].count).toBe(1));
-    it('Osd portability is disabled in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].portable).toBe(false));
+    cy.log('New device set is created').then(() =>
+      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length + 1),
+    );
+
+    cy.log('Device count is 1 in the new device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].count).toBe(1),
+    );
+
+    cy.log('Osd portability is disabled in the new device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].portable).toBe(false),
+    );
```

Comment:
This test is used in one other place as well. Perhaps we can create a helper function that tests if new device set is created and device count is added and portability is set as required.(use an args to test if it should be enabled/disabled).